xds/internal/xdsclient/xdslbregistry/converter/converter.go

```diff
@@ -90,6 +92,27 @@ func convertRingHashProtoToServiceConfig(rawProto []byte, depth int) (json.RawMe
 	return makeBalancerConfigJSON(ringhash.Name, rhCfgJSON), nil
 }
 
+type pfConfig struct {
+	ShuffleAddressList bool `json:"shuffleAddressList"`
+}
+
+func convertPickFirstProtoToServiceConfig(rawProto []byte, _ int) (json.RawMessage, error) {
+	if !envconfig.PickFirstLBConfig {
+		return nil, nil
+	}
+	pfProto := &v3pickfirstpb.PickFirst{}
+	if err := proto.Unmarshal(rawProto, pfProto); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal resource: %v", err)
+	}
+
+	pfCfg := &pfConfig{ShuffleAddressList: pfProto.GetShuffleAddressList()}
```

Comment:
With booleans, it's always best if `absent = false = default` is correct.  Sometimes with numbers, too.  But often with numbers `absent = some other default because zero is a bad or illegal, and requiring it to be set makes it annoying to use`.

Reply:
Right, as you mention below though this doesn't have the third option like OD number types so this doesn't apply here. But I see what you're saying about absent = false, and zero is bad or illegal e.g.